src/framework/MOM_file_parser.F90

```diff
@@ -1822,15 +1894,34 @@ subroutine get_param_real(CS, modulename, varname, value, desc, units, &
                                          !! multiplied by before it is returned.
   real,             optional, intent(out)   :: unscaled !< The value of the parameter that would be
                                          !! returned without any multiplication by a scaling factor.
+  character(len=*), optional, intent(in)    :: old_name !< A case-sensitive archaic name of the parameter
+                                         !! to read.  Errors or warnings are issued if the old name
+                                         !! is being used.
 
+  ! Local variables
   logical :: do_read, do_log
+  logical :: fatal_absence, new_name_used, old_name_used, same_value
+  real :: old_name_value  ! The value that is set when the old name is used.
 
   do_read = .true. ; if (present(do_not_read)) do_read = .not.do_not_read
   do_log  = .true. ; if (present(do_not_log))  do_log  = .not.do_not_log
 
   if (do_read) then
     if (present(default)) value = default
-    call read_param_real(CS, varname, value, fail_if_missing)
+    if (present(old_name)) then
+      old_name_value = value
+      call read_param_real(CS, old_name, old_name_value, set=old_name_used)
+      fatal_absence = .false. ; if ((.not.old_name_used) .and. present(fail_if_missing)) fatal_absence = fail_if_missing
+      call read_param_real(CS, varname, value, fatal_absence, set=new_name_used)
+
+      ! Issue appropriate warnings or error messages.
+      same_value = (new_name_used .and. old_name_used .and. (value == old_name_value))
+      if (old_name_used) call archaic_param_name_message(varname, old_name, new_name_used, same_value)
+
+      if (old_name_used .and. .not.new_name_used) value = old_name_value
+    else
+      call read_param_real(CS, varname, value, fail_if_missing)
+    endif
```

Comment:
I am really nitpicking here, but I think this section can be simplified as follows, which may help with readability.

```Fortran
    old_name_used = .false.

    if (present(old_name)) then
      new_name_value = value
      call read_param_real(CS, old_name, value, set=old_name_used)
      if (old_name_used) then
        call read_param_real(CS, varname, new_name_value, set=new_name_used)

        ! Issue appropriate warnings or error messages.
        same_value = (value == new_name_value)
        call archaic_param_name_message(varname, old_name, new_name_used, same_value)
      endif
    endif

    if (.not.old_name_used) then ! old_name is either not present or not set.
      call read_param_real(CS, varname, value, fail_if_missing)
    endif
```

Reply:
The suggested revisions would indeed be clearer and more parsimonious. I am modifying this PR as suggested.